template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreConfigProvider.java

```diff
@@ -101,8 +101,12 @@ public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor temp
 
         Optional<KerberosConfig> kerberosConfigOpt = source.getKerberosConfig();
         if (kerberosConfigOpt.isPresent()) {
-            String safetyValveValue = getSafetyValveProperty("hive.hook.proto.file.per.event", Boolean.TRUE.toString());
-            configs.add(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, safetyValveValue));
+            StringBuilder safetyValveValue = new StringBuilder(getSafetyValveProperty("hive.hook.proto.file.per.event", Boolean.TRUE.toString()));
+            if (source.getStackType() != null && source.getStackType().equals(StackType.DATALAKE)) {
+                safetyValveValue.append("<property><name>hive.metastore.try.direct.sql.ddl</name><value>true</value></property>");
+                safetyValveValue.append("<property><name>hive.metastore.try.direct.sql</name><value>true</value></property>");
+            }
```

Comment:
What is the rationale behind moving this out from the bp and adding it dynamically?